<h2 id="power-for-de-novo-and-casecontrol-studies">Power for de novo and case/control studies</h2>

<p>Hailiang Huang<sup>1</sup>, Benjamin Neale<sup>1</sup> and Stephan Sanders<sup>2</sup> <br>
1 Analytic and Translational Genetics Unit, Massachusetts General Hospital and the Broad Institute of MIT and Harvard<br>
2 Department of Psychiatry, School of Medicine, University of California San Francisco</p>



<h3 id="power-for-de-novo-analysis">Power for de novo analysis</h3>

<p>Assume the expected number of de novo mutations per individual genome is <script type="math/tex" id="MathJax-Element-1">Q_0</script>.  If we perform the de novo analysis on a proportion of the genome (denoted as <script type="math/tex" id="MathJax-Element-2">f</script>, <script type="math/tex" id="MathJax-Element-3">f\in[0,1]</script>), the expected number of mutations in this selected region of genome is <script type="math/tex" id="MathJax-Element-4">fQ_0</script>. <br>
For cases, assume a proportion (denoted as <script type="math/tex" id="MathJax-Element-5">p</script>, <script type="math/tex" id="MathJax-Element-6">p\in[0,1]</script>) of the de novo mutations are functional with with relative risk of <script type="math/tex" id="MathJax-Element-7">R</script>. The expected number of de novo mutations in cases is (see proof 1): <script type="math/tex" id="MathJax-Element-8">Q_1=(1-p)fQ_0+RpfQ_0</script> .  </p>



<h4 id="simulation">Simulation</h4>

<p>We simulate the number of de novo mutations for contols using a poisson distribution with the poission parameter of <script type="math/tex" id="MathJax-Element-9">fQ_0</script>.  We simulate the number of de novo mutations for cases using a mixture of two poisson distributions with poisson parameters of <script type="math/tex" id="MathJax-Element-10">(1-p)fQ_0</script> and <script type="math/tex" id="MathJax-Element-11">RpfQ_0</script> respectively. We then perform a poisson regression and get the deviance between the poisson model (simulated number of de novo mutations versus case/control status) and the null model. The deviance is chi-square distributed and will be used to calculate the power. This process is repeated several times. </p>



<h4 id="parametric">Parametric</h4>

<p>When the sample size is large,  poisson test can be approximiated using the t test.  The chi-square distributed test statistic can be calculated as (see proof 2): <br>
<script type="math/tex; mode=display" id="MathJax-Element-12">\chi^2=\frac{(Q_1-Q_0)^2}{Q_1/N_1+Q_0/N_0},</script>  <br>
in which <script type="math/tex" id="MathJax-Element-13">N_1</script> and <script type="math/tex" id="MathJax-Element-14">N_0</script> are the sample size for case and controls respectively. </p>



<h3 id="power-for-casecontrol-analysis">Power for case/control analysis</h3>

<p>Let <script type="math/tex" id="MathJax-Element-15">x_i</script> be the genotype of variant <script type="math/tex" id="MathJax-Element-16">i</script> and <script type="math/tex" id="MathJax-Element-17">y</script> be the phenotype of all individuals.  We assume there are <script type="math/tex" id="MathJax-Element-18">s</script> SNPs in a gene and <script type="math/tex" id="MathJax-Element-19">f</script> of them are functional.  The allele frequency of the SNPs is sampled from the exponential distribution: <script type="math/tex" id="MathJax-Element-20">\Pr(AF)=\lambda\exp ^ {-\lambda AF}</script> with <script type="math/tex" id="MathJax-Element-21">\lambda=1/\overline{AF}</script> (the sampled AF therefore has mean of <script type="math/tex" id="MathJax-Element-22">\overline{AF}</script>). The probability for being a causal variant scales with the allele frequency: <script type="math/tex" id="MathJax-Element-23">\Pr(\mathrm{causal}|AF)=1/AF</script>.</p>



<h4 id="single-variant-test">Single variant test</h4>

<p>The power for a single variant can be calculated using the chi-square distribution with the non-centrality parameter <script type="math/tex" id="MathJax-Element-24">\chi^2</script>. This parameter is estimated using the ‘NCPgen()’ function in the source code (basically a 2x2 contingency table test). The power for finding at least one out of the <script type="math/tex" id="MathJax-Element-25">f</script> functional SNPs is <script type="math/tex" id="MathJax-Element-26">1-\prod_{i \in f}(1-\mathrm{power_{snp_i}})</script></p>



<h4 id="burden-test">Burden test</h4>

<p>In burden test, assuming SNPs are independent, we have <script type="math/tex" id="MathJax-Element-27">x = \sum_{i \in f}x_i + \sum _ {j\in s-f} x_j </script>. The coefficients are <br>
<script type="math/tex; mode=display" id="MathJax-Element-28">
\begin{aligned}
\beta' & = \frac{cov(y,x)}{var(x)} \\
&=\frac{\sum _ {i\in f} cov(x_i, y) + \sum _ {j\in s-f} cov(x_j, y)}{\sum _ {i\in f} var(x_i) + \sum_{j\in s-f} var(x_j)} \\
&=\frac{\sum _ {i\in f} cov(x_i, y) + \sum _ {j\in s-f} cov(x_j, y)}{\sum _ {i\in f} var(x_i) + \sum _ {j\in s-f} var(x_j)} \\
&=\frac{\beta \sum _ {i\in f}  var(x_i) + 0* \sum _ {j\in s-f} var(x_j)}{\sum _ {i\in f} var(x_i) + \sum _ {j\in s-f} var(x_j)}\\
&=\beta \frac{\sum _ {i\in f}  var(x_i) }{\sum _ {i\in f} var(x_i) + \sum _ {j\in s-f} var(x_j)}\\
\end{aligned}
</script></p>

<p>The power for burden test can be calculated using the chi-square distribution with the non-centrality parameter <script type="math/tex" id="MathJax-Element-29">\chi^2</script> calculated using the new coefficient <script type="math/tex" id="MathJax-Element-30">\beta'</script>. </p>



<h3 id="proofs">Proofs</h3>



<h4 id="proof-1">Proof 1</h4>

<p>We prove that under the conditions a) the population prevalence of the disease is small (<script type="math/tex" id="MathJax-Element-31">\Pr(case)\ll1</script>) and b) the de novo rate is low (<script type="math/tex" id="MathJax-Element-32">\Pr(A)\equiv\mu\ll1</script>), <script type="math/tex; mode=display" id="MathJax-Element-33">\frac{\Pr(A|case)}{\Pr(A|control)} \simeq R,</script> <br>
in which <script type="math/tex" id="MathJax-Element-34">R</script> is the relative risk. Recall that <script type="math/tex" id="MathJax-Element-35">R\equiv\frac{\Pr(case|A)}{\Pr(case|a)}</script> <br>
<script type="math/tex; mode=display" id="MathJax-Element-36">
\begin{aligned}
\frac{\Pr(A|case)}{\Pr(A|control)} &=&\frac{\Pr(case|A)}{\Pr(control|A)}\times\frac{\Pr(control)}{\Pr(case)}\\
&=&R\times\frac{\Pr(case|a)}{\Pr(control|A)}\times\frac{\Pr(control)}{\Pr(case)}\\
&=&R\times\frac{\Pr(case|a)}{\Pr(case)}\times\frac{\Pr(control)}{\Pr(control|A)}. 
\end{aligned}\tag{0}
</script> <br>
We will then prove <script type="math/tex" id="MathJax-Element-37">\frac{\Pr(case|a)}{\Pr(case)}\simeq 1</script> and <script type="math/tex" id="MathJax-Element-38">\frac{\Pr(control)}{\Pr(control|A)}\simeq 1</script>. <br>
<script type="math/tex; mode=display" id="MathJax-Element-39">
\begin{aligned}
\frac{\Pr(case|a)}{\Pr(case)}&=&\frac{\Pr(case|a)}{\Pr(case|a)\Pr(a)+\Pr(case|A)\Pr(A)} \\
&=&(\Pr(a)+\frac{\Pr(case|A)}{\Pr(case|a)}\mu)^{-1} \\
&=&(\Pr(a)+R\mu)^{-1} \\
&=&(1-\mu+R\mu)^{-1} \\
&=&(1+(R-1)\mu)^{-1} \\
&\simeq&1-(R-1)\mu \\
&\simeq&1.
\end{aligned}\tag{1}
</script> <br>
And  <br>
<script type="math/tex; mode=display" id="MathJax-Element-40">
\begin{aligned}  
\frac{\Pr(control)}{\Pr(control|A)}&=&\frac{\Pr(control)}{1-\Pr(case|A)}\\
&=&\frac{\Pr(control)}{1-R\Pr(case|a)}.  
\end{aligned}\tag{2}
</script> <br>
For <script type="math/tex" id="MathJax-Element-41">\Pr(case|a)</script>, we have <br>
<script type="math/tex; mode=display" id="MathJax-Element-42">
\begin{aligned}
\Pr(case|a)&=&\frac{\Pr(case)}{\Pr(a)+R\Pr(A)}\\
&=&\frac{\Pr(case)}{1+(R-1)\mu}\\
&\simeq&K-K(R-1)\mu.
\end{aligned}\tag{3}
</script> <br>
Using equations 2 and 3, we have <br>
<script type="math/tex; mode=display" id="MathJax-Element-43">
\begin{aligned}  
\frac{\Pr(control)}{\Pr(control|A)}&\simeq&\frac{\Pr(control)}{1-R(K-K(R-1)\mu)} \\ 
&\simeq&\frac{1-K}{1-RK+R(R-1)K\mu} \\
&\simeq&(1-K)(1+RK-R(R-1)K\mu) \\
&\simeq&1+(R-1)K-RK^2\\
&& -R(R-1)K\mu+R(R-1)K^2\mu \\
&\simeq&1.
\end{aligned} \tag{4}
</script> <br>
Using equations 0, 1 and 4: <script type="math/tex" id="MathJax-Element-44">\frac{\Pr(A|case)}{\Pr(A|control)}\simeq R</script>.</p>



<h4 id="proof-2">Proof 2</h4>

<p>We prove that the following is chi-square distributed: <script type="math/tex; mode=display" id="MathJax-Element-45">\chi^2=\frac{(Q_1-Q_0)^2}{Q_1/N_1+Q_0/N_0}</script> . <br>
We know the following quantity from the <script type="math/tex" id="MathJax-Element-46">t</script> test follows a normal distribution <br>
<script type="math/tex; mode=display" id="MathJax-Element-47">z=\frac{\bar{x}_1-\bar{x}_2}{\sqrt{var(x_1)/N_1+var(x_2)/N_2}}</script> . <br>
We assume group 1 is the case cohort and group 2 is the control cohort.  We know that the poisson parameter is same as the mean and the variance of a poisson distribution.  Therefore, <script type="math/tex" id="MathJax-Element-48">\bar{x}_1=Q_1</script>  and <script type="math/tex" id="MathJax-Element-49">\bar{x}_2=Q_0</script>; <script type="math/tex" id="MathJax-Element-50">var(x_1)=(1-p)fQ_0+RpfQ_0=Q_1</script> (assumes the two poisons in the mixture are independent) and <script type="math/tex" id="MathJax-Element-51">var(x_2)=Q_0</script>.  <script type="math/tex" id="MathJax-Element-52">\chi^2</script> is simply <script type="math/tex" id="MathJax-Element-53">z^2</script>. </p>